src/shared/VacationTable/VacationTable.tsx

```diff
@@ -137,6 +140,31 @@ const VacationTable: FC = () => {
     setPage(0);
   };
 
+  const handleEdit = (id: string): void => {
+    setEdit(id);
+    navigate("/vacation");
+  };
+
+  const deleteVacation = (id: string) => {
+    const getData = JSON.parse(localStorage.getItem("data") as string);
+    const newVacation: IDataVacation[] = data.filter(
+      (vacation) => vacation.id !== id
+    );
+
+    localStorage.setItem(
+      "data",
+      JSON.stringify({
+        ...getData,
+        [user?.email as string]: [...newVacation],
```

Comment:
I guess at this port user shouldn't be possibly undefined, because in this case, you will save data for the undefined user. Which means data is lost.